backend/controllers/userMe.test.js

```diff
@@ -31,35 +28,113 @@ describe('userMe controller', () => {
     keyworkerApi.getPrisonMigrationStatus = jest.fn()
 
     oauthApi.currentUser.mockImplementation(() => staff1)
+    oauthApi.userRoles.mockImplementation(() => [])
     elite2Api.userCaseLoads.mockImplementation(() => caseloads)
     keyworkerApi.getPrisonMigrationStatus.mockImplementation(() => ({
       migrated: true,
     }))
     res.json = jest.fn()
   })
 
-  it('should not have writeAccess when the user does not have the key worker admin role', async () => {
-    oauthApi.userRoles.mockImplementation(() => [])
-    keyworkerApi.getPrisonMigrationStatus.mockImplementation(() => ({
-      migrated: false,
-      supported: false,
-      autoAllocatedSupported: false,
-      kwSessionFrequencyInWeeks: 1,
-      capacityTier1: 1,
-      capacityTier2: 2,
-    }))
-    const { userMeService } = userMeFactory(oauthApi, elite2Api, keyworkerApi)
-    await userMeService(req, res)
-
-    expect(res.json).toBeCalledWith({
+  describe('access checks', () => {
+    const defaultUserMe = {
       ...staff1,
       activeCaseLoadId: 'LEI',
+      prisonMigrated: true,
+      maintainAccess: false,
+      maintainAccessAdmin: false,
+      migration: false,
       writeAccess: false,
-      prisonMigrated: false,
+      maintainAuthUsers: false,
+    }
+
+    it('should default to no access if user has no roles', async () => {
+      await userMeService(req, res)
+
+      expect(res.json.mock.calls[0][0]).toEqual({
+        ...defaultUserMe,
+      })
+    })
+    it('should have writeAccess when the user has the key worker admin role', async () => {
+      oauthApi.userRoles.mockImplementation(() => [{ roleCode: 'OMIC_ADMIN' }])
+      await userMeService(req, res)
+
+      expect(res.json.mock.calls[0][0]).toEqual({
```

Comment:
should be able to use expect(res.json).toHaveBeenCalledWith({
       ...defaultUserMe,
        writeAccess: true,
      })